Intersect.Client/Entities/Player.cs

```diff
@@ -2029,6 +2147,54 @@ private void ProcessDirectionalInput()
                                 OffsetX = -Options.TileWidth;
                             }
 
+                            break;
+                        case (byte)Directions.UpLeft:
+                            if (IsTileBlocked(X - 1, Y - 1, Z, MapId, ref blockedBy) == -1)
+                            {
+                                tmpY--;
+                                tmpX--;
+                                IsMoving = true;
+                                Dir = 4;
+                                OffsetY = Options.TileHeight;
+                                OffsetX = Options.TileWidth;
+                            }
+
+                            break;
+                        case (byte)Directions.UpRight:
+                            if (IsTileBlocked(X + 1, Y - 1, Z, MapId, ref blockedBy) == -1)
+                            {
+                                tmpY--;
+                                tmpX++;
+                                IsMoving = true;
+                                Dir = 5;
+                                OffsetY = Options.TileHeight;
+                                OffsetX = -Options.TileWidth;
+                            }
+
+                            break;
+                        case (byte)Directions.DownLeft:
+                            if (IsTileBlocked(X - 1, Y + 1, Z, MapId, ref blockedBy) == -1)
+                            {
+                                tmpY++;
+                                tmpX--;
+                                IsMoving = true;
+                                Dir = 6;
+                                OffsetY = -Options.TileHeight;
+                                OffsetX = Options.TileWidth;
+                            }
+
+                            break;
+                        case (byte)Directions.DownRight:
+                            if (IsTileBlocked(X + 1, Y + 1, Z, MapId, ref blockedBy) == -1)
+                            {
+                                tmpY++;
+                                tmpX++;
+                                IsMoving = true;
+                                Dir = 7;
+                                OffsetY = -Options.TileHeight;
+                                OffsetX = -Options.TileWidth;
+                            }
+
```

Comment:
You can change the value of the all Dir variables, also for (byte)Enums
from L2107 until L2197